vignettes/quick-start.Rmd

```diff
@@ -0,0 +1,161 @@
+---
+title: "Quick Start"
+author: "R Validation Hub"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Quick Start}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+library(riskmetric)
+library(dplyr)
+library(tibble)
+
+options(repos = "https://cran.rstudio.com")
+
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+Whether we're looking for a robust choice among a set of similar packages,
+require strong development and maintainence practices for package reliability,
+or need to prove the validity of packages in a regulated industry, we will
+follow the same sequence of steps.
+
+1. **Find a suitable reference for the package**  
+   this can be any source of package metadata, such as a
+    * package repository webportal from which we can scrape metadata
+    * built and installed package in a local library
+    * directory containing package source code 
+1. **Assess package under validation criteria**  
+   collect a set of metadata for scoring risk, this might include
+    * a profile of the number of active users over the last month
+    * the availability of documentation for package functions
+    * the use of a unit testing framework
+    * the test coverage
+1. **Score all of the assessed criteria**  
+1. **Summarize scores into an aggregate risk metric**  
+
+
+
+# Package Reference Objects
+
+Before we get started assessing packages, we need a place to aggregate all of
+our package metadata. To handle this, we use the `pkg_ref` class. This class is
+an environment to allow for by-reference mutability so that metadata necessary
+to derive one metric can be reused by another without going out and fetching
+data all over again.
+
+```{r, eval = FALSE}
+library(riskmetric)
+pkg_ref("riskmetric")
+```
+
+```{r, echo = FALSE}
+rver <- gsub("\\.\\d+$", "", paste0(R.version$major, ".", R.version$minor))
+package <- pkg_ref("riskmetric")
+
+# hack in order to mutate package environment directly so nobody accidentally
+# publishes any personal info in their library path
+invisible(riskmetric:::bare_env(package, {
+  package$path <- sprintf("/home/user/username/R/%s/Resources/library/riskmetric", rver)
+}))
+
+package
+```
+
+Likewise, `pkg_ref` objects can be coerced to a `tibble` for easier,
+parallel assessment of packages.
+
+```{r, eval = FALSE}
+library(dplyr)
+package_tbl <- pkg_ref(c("riskmetric", "utils", "survival")) %>%
+  as_tibble()
+```
+
+```{r, echo = FALSE}
+package_tbl <- pkg_ref(c("riskmetric", "utils", "survival")) %>%
+  as_tibble() %>%
+  tibble:::print.tbl()
+```
+
+
+
+# Core Verbs
+
+In `riskmetric`, the core operations of package validation are handled in a
+series of verbs to allow easy chaining of operations. These verbs, as well as
```

Comment:
suggest to add one sentence to summary verbs that corresponding to each step.

Reply:
This happens in each of the subsequent sections. 

https://github.com/pharmaR/riskmetric/blob/03ebf2bbaf5d8651ad034f41a7e72abac5f9fc89/vignettes/quick-start.Rmd#L107-L111

https://github.com/pharmaR/riskmetric/blob/03ebf2bbaf5d8651ad034f41a7e72abac5f9fc89/vignettes/quick-start.Rmd#L125-L128

https://github.com/pharmaR/riskmetric/blob/03ebf2bbaf5d8651ad034f41a7e72abac5f9fc89/vignettes/quick-start.Rmd#L136-L142

I think adding similar sentences to the intro would make this section repeat itself too much.